packages/gatsby/src/utils/worker/pool.ts

```diff
@@ -69,7 +69,29 @@ export async function runQueriesInWorkersQueue(
   }
 
   await Promise.all(promises)
+  await Promise.all(pool.all.saveQueries())
```

Comment:
Something to consider:

If we await `Promise.all(pool.all.saveQueries())` we could drop await from `await Promise.all(promises)`. This would make "idle" worker immediatelly persist once they are done with all query running instead of each worker waiting for all workers to finish.